src/repositories/schemas.ts

```diff
@@ -1,9 +1,195 @@
 import { OAuth1Client, ObjectId } from '@extrahorizon/javascript-sdk';
 
+interface Transition {
+  name: string;
+  type: string;
+  fromStatuses: string[];
+  toStatus: string;
+  conditions: { type: string; };
+  actions: { type: string; };
+  afterActions: { type: string; };
+}
+
 export async function remove(sdk: OAuth1Client, schemaId: ObjectId) {
   return await sdk.data.schemas.remove(schemaId);
 }
 
 export async function disable(sdk: OAuth1Client, schemaId: ObjectId) {
   return await sdk.data.schemas.disable(schemaId);
 }
+
+/**
+ * fetch a full schema from the service from a name
+ * @param {string} name the name of a schema
+ */
+export async function fetchSchemaByName(sdk: OAuth1Client, name: string) {
+  return await sdk.data.schemas.findByName(name);
+}
+
+/**
+ * create a schema using the data service
+ * @param {string} name         name of the desired schema
+ * @param {string} description  description of the schema
+ */
+export async function createSchema(sdk: OAuth1Client, name: string, description: string) {
+  return await sdk.data.schemas.create({
+    name,
+    description,
+  });
+}
+
+/**
+ * update root attributes of a schema
+ * @param {string} id   the schema identifier
+ * @param {object} data the target schema
+ * @param {string} data.description
+ * @param {number} data.defaultLimit
+ * @param {number} data.maximumLimit
+ */
+export async function updateSchema(sdk: OAuth1Client, id: string, data: any) {
+  return await sdk.data.schemas.update(id, data);
+}
+
+/**
+ * create a property
+ * @param {string} id   the schema identifier
+ * @param {object} data a property object
+ * @param {string} data.name
+ * @param {object} data.configuration
+ * @param {string} data.configuration.type
+ */
+export async function createProperty(sdk: OAuth1Client, id: string, data: { name: string; configuration: { type: string; }; }) {
```

Comment:
This contract is quite wrong no? A property is much more than just a type? 🤔

Reply:
Yes it is not complete at all. But that was already like that before and it does not break anything because it is just the type. I will make a ticket for this.